.github/labeler.yml

```
@@ -0,0 +1,30 @@
+stm32:
+  - chips/stm*/**/*
+
+nrf:
+  - chips/nrf5*/**/*
+
+sam4l:
+  - chips/sam4l/**/*
+
+hil:
+  - kernel/src/hil/*
+
+risc-v:
+  - arch/rv32i/**/*
+  - arch/riscv/**/*
+
+tock-libraries:
+  - libraries/**/*
+
+WG-OpenTitan:
+  - boards/earlgrey-nexysvideo/**/*
+  - chips/earlgrey/**/*
+
+# add kernel label unless already covered by hil label
+kernel:
+  - any: ['kernel/**/*', '!kernel/src/hil/*']
+
+# add documentation label only if all changes are in doc/, or in md or txt files
+documentation:
+  - all: ['doc/*', './**/*.md', './**/*.txt']
```

```suggestion
  - all: ['doc/*']
```

Do we have .txt files? Also, pull requests like new boards touch markdown files, but I wouldn't consider those documentation PRs.

---

a chunk of the working group notes have been uploaded as `.txt` files. And the use of the 'all' rule means that the `documentation` label is only added if *every* changed file matches *one* of the globs in that rule. So new board PRs would not get the label because they change `*.rs` files.

---

actually, the docs say: "`all`: match ALL globs against ALL changed paths". So this needs to be updated in order to get the behavior I just described.

---

Hey at least I'm consistent! I could see:

```
- all: ['doc/*']
```

or
```
- any: ['doc/*']
- all: ['./**/*.md']
```
as reasonable rules for this.

---

I just went with the simple approach of requiring all changes be in `doc/`